components/browser/icons/src/main/java/mozilla/components/browser/icons/IconRequest.kt

```diff
@@ -20,7 +20,8 @@ data class IconRequest(
     val url: String,
     val size: Size = Size.DEFAULT,
     val resources: List<Resource> = emptyList(),
-    @ColorInt val color: Int? = null
+    @ColorInt val color: Int? = null,
+    val isPrivate: Boolean = false
```

Comment:
@pocmo general question: What's the policy on API data model changes? For instance, in this IconRequest object, if a new mandatory uninitialized property is added (in this specific scenario if I had omitted the false value for isPrivate), then all callers of the constructor IconRequest() (even in other modules) would have to change. My understanding of reusable components makes me think that data model changes are forbidden. But, wanted your thoughts on it.

Reply:
They are not forbidden but yeah we try to avoid them if possible, either by adding a default value, like you did or by adding the new API and deprecating the old API. If we can't avoid it then we make the breaking change and explicitly mention that in the cchangelog.